dec_rgb.c

```diff
@@ -18,7 +18,7 @@
 int fswc_add_image_rgb32(src_t *src, avgbmp_t *abitmap)
 {
 	uint8_t *img = (uint8_t *) src->img;
-	uint32_t i = src->width * src->height;
+	uint32_t i = src->width * src->height * 4;
```

Comment:
This function counts samples not bytes, so this is wrong. I think the best solution would be to change this to match the way `fswc_add_image_bgr32()` works.

You'll also need to remove the `<< 2` in the subsequent lines in both functions.

Reply:
This means the fix I made for fswc_add_image_bgr32() is good ?

Reply:
Almost, the  `<< 2` needs to be removed in `fswc_add_image_bgr32()`

Reply:
So just to be clear, the original `fswc_add_image_rgb32()` was fine, you need to revert your change to that. Your fix for `fswc_add_image_bgr32()` is correct but incomplete, just needs that shift removed.